src/controllers/csv-processor.ts

```diff
@@ -161,55 +179,30 @@ export const moveCSVFromBlobStorageToDatalake = async (dataset: Dataset): Promis
     }
 };
 
-export const uploadCSVToBlobStorage = async (buff: Buffer, dataset: Dataset): Promise<UploadDTO | UploadErrDTO> => {
+export const uploadCSVToBlobStorage = async (fileStream: Readable, filetype: string): Promise<Import> => {
     const blobStorageService = new BlobStorageService();
-    if (buff) {
-        const hash = createHash('sha256').update(buff).digest('hex');
-        const datafile = Datafile.createDatafile(dataset, hash.toString(), 'BetaUser');
-        const savedDataFile = await datafile.save();
-        const dto = await datasetToDatasetDTO(dataset);
+    if (fileStream) {
+        const importRecord = new Import();
+        importRecord.id = randomUUID();
+        importRecord.mime_type = filetype;
         try {
-            logger.debug(`Uploading file ${savedDataFile.id} to blob storage`);
-            await blobStorageService.uploadFile(`${savedDataFile.id}.csv`, buff);
-            return {
-                success: true,
-                dataset: dto
-            };
+            await blobStorageService.uploadFile(`${importRecord.id}.csv`, fileStream);
+            const resolvedHash = await hashReadableStream(fileStream)
```

Comment:
So if the blobStorage request fails it throws an error... This sits in a try catch box for this reason... If it errors we never make it to the point of generating the hash.  We can only generate the has if we have successfully uploaded the file to Blob Storage.

Reply:
The problem isn't just when it throws an error, it's when it simply takes longer to process data than the digest function (which should be always as the digest is some simple local compute and the blobstore is some networked IO).

We need to generate the hash *as* we upload it so that we get the benefit of the streaming interface: only having one block in memory at a time.